### PR TITLE
fix: increase the vault health retry timeout to 10 minutes and add some more logging

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/kube/vault/vault_factory.go [hunk 1]
```diff
@@ -167,6 +169,7 @@ func waitForVault(vaultClient *api.Client, initialDelay, timeout time.Duration)
 func waitForKVEngine(vaultClient *api.Client, initialDelay, timeout time.Duration) error {
 	return util.RetryWithInitialDelaySlower(initialDelay, timeout, func() error {
 		if _, err := vaultClient.Logical().Read(kvEngineConfigPath); err != nil {
+			log.Logger().Info("Waiting for KV engine to be configured...")
```
Comment (pmuir) [inline pkg/kube/vault/vault_factory.go]: This doesn't seem right - it's just before returning an error

Comment (ccojocar) [reply pkg/kube/vault/vault_factory.go]: The retry takes place on error. This log is intended to show some action when a new retry is being performed.

Comment (pmuir) [reply pkg/kube/vault/vault_factory.go]: oic

**Pull request merged**